src/Npgsql/Replication/NpgsqlReplicationConnection.cs

```diff
@@ -0,0 +1,891 @@
+
+        ReplicationConnectionState State
+        {
+            get => (ReplicationConnectionState)_state;
+            set
+            {
+                // Disposed and Broken are final states. We never leave them.
+                if (State == ReplicationConnectionState.Disposed || State == ReplicationConnectionState.Broken)
+                    return;
+
+                _state = (int)value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the wait time before terminating the attempt  to execute a command and generating an error.
+        /// </summary>
+        /// <value>The time to wait for the command to execute. The default value is 30 seconds.</value>
+        public TimeSpan CommandTimeout
+        {
+            get => _commandTimeout ?? (_npgsqlConnection.CommandTimeout > 0
+                ? TimeSpan.FromSeconds(_npgsqlConnection.CommandTimeout)
+                : Timeout.InfiniteTimeSpan);
+            set
+            {
+                if (value < TimeSpan.Zero && value != Timeout.InfiniteTimeSpan)
+                    throw new ArgumentOutOfRangeException(nameof(value), value,
+                        $"A finite CommandTimeout can't be less than {TimeSpan.Zero}.");
+
+                _commandTimeout = value;
+                if (State != ReplicationConnectionState.Streaming)
+                    SetTimeouts(value, value);
+            }
+        }
+
+        /// <summary>
+        /// The client encoding for the connection
+        /// This can only be called when there is an active connection.
+        /// </summary>
+        public Encoding Encoding => _npgsqlConnection.Connector?.TextEncoding ?? throw new InvalidOperationException($"The {Encoding} property can only be used when there is an active connection");
+
+        /// <summary>
+        /// Process id of backend server.
+        /// This can only be called when there is an active connection.
+        /// </summary>
+        public int ProcessID => _npgsqlConnection.Connector?.BackendProcessId ?? throw new InvalidOperationException($"The {ProcessID} property can only be used when there is an active connection");
+
+        #endregion Properties
+
+        #region OpenAsync / DisposeAsync
+
+        /// <summary>
+        /// Opens a database replication connection with the property settings specified by the
+        /// <see cref="NpgsqlReplicationConnection.ConnectionString">ConnectionString</see>.
+        /// </summary>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.
+        /// The default value is <see cref="CancellationToken.None"/>.</param>
+        /// <returns>A task representing the asynchronous open operation.</returns>
+        public async Task OpenAsync(CancellationToken cancellationToken = default)
```

Comment:
We need to review what we do with OpenAsync - I think we're inconsistent. We have OpenAsync, but StartReplication (without Async).

Reply:
Yeah, let's remove the `Async` suffix. We're async all over the place without suffix anyways.

Reply:
I tend to agree, am just uneasy with changing OpenAsync to Open given that we already have NpgsqlConnection with both Open and OpenAsync.

Reply:
Look at this again, I've overcome my uneasiness and will rename OpenAsync to Open :)

Reply:
Done in cba33745977d3023d35b0602cbff8889191a6da2